pkg/utils/tekton/controller.go

```diff
@@ -1,835 +1,17 @@
-				Name: "skopeo-copy",
-			},
-			Params: []v1beta1.Param{
-				{
-					Name: "srcImageURL",
-					Value: v1beta1.ParamValue{
-						StringVal: srcImageURL,
-						Type:      v1beta1.ParamTypeString,
-					},
-				},
-				{
-					Name: "destImageURL",
-					Value: v1beta1.ParamValue{
-						StringVal: destImageURL,
-						Type:      v1beta1.ParamTypeString,
-					},
-				},
-			},
-			// workaround to avoid the error "container has runAsNonRoot and image will run as root"
-			PodTemplate: &pod.Template{
-				SecurityContext: &corev1.PodSecurityContext{
-					RunAsNonRoot: pointer.Bool(true),
-					RunAsUser:    pointer.Int64(65532),
-				},
-			},
-			Workspaces: []v1beta1.WorkspaceBinding{
-				{
-					Name:     "images-url",
-					EmptyDir: &corev1.EmptyDirVolumeSource{},
-				},
-			},
-		},
-	}
-
-	err := s.KubeRest().Create(context.TODO(), &taskRun)
-	if err != nil {
-		return nil, err
-	}
-	return &taskRun, nil
-}
-
-// GetTaskRun returns the requested TaskRun object
-func (s *SuiteController) GetTaskRun(name, namespace string) (*v1beta1.TaskRun, error) {
-	namespacedName := types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
-	}
-
-	taskRun := v1beta1.TaskRun{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-		},
-	}
-	err := s.KubeRest().Get(context.TODO(), namespacedName, &taskRun)
-	if err != nil {
-		return nil, err
-	}
-	return &taskRun, nil
-}
-
-// CreateSkopeoCopyTask creates a skopeo copy task in the given namespace
-func (s *SuiteController) CreateSkopeoCopyTask(namespace string) error {
-	_, err := exec.Command(
-		"oc",
-		"apply",
-		"-f",
-		"https://api.hub.tekton.dev/v1/resource/tekton/task/skopeo-copy/0.2/raw",
-		"-n",
-		namespace).Output()
-
-	return err
-}
-
-// Remove all Tasks from a given repository. Useful when creating a lot of resources and wanting to remove all of them
-func (h *SuiteController) DeleteAllTasksInASpecificNamespace(namespace string) error {
-	return h.KubeRest().DeleteAllOf(context.TODO(), &v1beta1.Task{}, crclient.InNamespace(namespace))
-}
-
-// Remove all TaskRuns from a given repository. Useful when creating a lot of resources and wanting to remove all of them
-func (h *SuiteController) DeleteAllTaskRunsInASpecificNamespace(namespace string) error {
-	return h.KubeRest().DeleteAllOf(context.TODO(), &v1beta1.TaskRun{}, crclient.InNamespace(namespace))
-}
-
-// GetTask returns the requested Task object
-func (s *SuiteController) GetTask(name, namespace string) (*v1beta1.Task, error) {
-	namespacedName := types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
-	}
-
-	task := v1beta1.Task{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-		},
-	}
-	err := s.KubeRest().Get(context.TODO(), namespacedName, &task)
-	if err != nil {
-		return nil, err
-	}
-	return &task, nil
+func NewSuiteController(kube *kubeCl.CustomClient) (*TektonController, error) {
+	return &TektonController{
+		kube,
+	}, nil
```

Comment:
probably no reason to have `error` type of return value, if it always returns `nil`

Reply:
Removed.